Occasional flaky tests with grouped output.

### Description

There is an occasional flaky test result. It seems like the output of task is captured with split lines, that is to say, a partial line is written as one line, and then the rest of that line is written as a subsequent line. This output, when sorted, the causes the goldie check to fail.

For instance, in the following example, the single line "d3\n" seems to be "d3" and "\n". As a result the check fails.

```
--- FAIL: TestDeps (0.14s)
      executor_test.go:401: Result did not match the golden fixture. Diff is below:
          
          --- Expected
          +++ Actual
          @@ -1,2 +1,2 @@
          -d1
          +
           d11
          @@ -8,3 +8,2 @@
           d23
          -d3
           d31
          @@ -12,2 +11,3 @@
           d33
          +d3d1
           
          
  FAIL
  FAIL	github.com/go-task/task/v3	6.995s
```

Its possible that the function https://github.com/go-task/task/blob/e72c35f79ffd32856efe3226ff9aadcd78d0804c/task_test.go#L304 is not processing the output correctly. It may first need to join lines without a trailing '\n' to the following line, and then apply the sorting algorithm. However, this is not proven, and the output "d3d1\n" does seem odd. In theory, with grouped output, such a combination should not be possible, right? Therefore I wonder if the post processing (`PPSortedLines()`) causes the issue.

I observe this problem most often with the "deps" test: https://github.com/go-task/task/blob/e72c35f79ffd32856efe3226ff9aadcd78d0804c/testdata/deps/Taskfile.yml#L1

### Version

v3.44.1

### Operating system

Ubuntu

### Experiments Enabled

_No response_

### Example Taskfile

```YAML

```

--- a/executor_test.go
+++ b/executor_test.go
@@ -144,12 +144,12 @@ func (tt *ExecutorTest) run(t *testing.T) {
 	t.Helper()
 	f := func(t *testing.T) {
 		t.Helper()
-		var buf bytes.Buffer
+		var buffer SyncBuffer
 
 		opts := append(
 			tt.executorOpts,
-			task.WithStdout(&buf),
-			task.WithStderr(&buf),
+			task.WithStdout(&buffer),
+			task.WithStderr(&buffer),
 		)
 
 		// If the test has input, create a reader for it and add it to the
@@ -172,7 +172,7 @@ func (tt *ExecutorTest) run(t *testing.T) {
 		if err := e.Setup(); tt.wantSetupError {
 			require.Error(t, err)
 			tt.writeFixtureErrSetup(t, g, err)
-			tt.writeFixtureBuffer(t, g, buf)
+			tt.writeFixtureBuffer(t, g, buffer.buf)
 			return
 		} else {
 			require.NoError(t, err)
@@ -193,7 +193,7 @@ func (tt *ExecutorTest) run(t *testing.T) {
 		if err := e.Run(ctx, call); tt.wantRunError {
 			require.Error(t, err)
 			tt.writeFixtureErrRun(t, g, err)
-			tt.writeFixtureBuffer(t, g, buf)
+			tt.writeFixtureBuffer(t, g, buffer.buf)
 			return
 		} else {
 			require.NoError(t, err)
@@ -206,7 +206,7 @@ func (tt *ExecutorTest) run(t *testing.T) {
 			}
 		}
 
-		tt.writeFixtureBuffer(t, g, buf)
+		tt.writeFixtureBuffer(t, g, buffer.buf)
 	}
 
 	// Run the test (with a name if it has one)

--- a/internal/output/group.go
+++ b/internal/output/group.go
@@ -44,10 +44,15 @@ func (gw *groupWriter) close() error {
 		// don't print begin/end messages if there's no buffered entries
 		return nil
 	}
-	if _, err := io.WriteString(gw.writer, gw.begin); err != nil {
-		return err
+	if len(gw.begin) > 0 {
+		// Rewrite the gw.buff with the beginning text.
+		s := gw.buff.String()
+		gw.buff.Reset()
+		gw.buff.WriteString(gw.begin)
+		gw.buff.WriteString(s)
 	}
 	gw.buff.WriteString(gw.end)
+	// Return the entire gw.buff to ensure the group is written atomically to stdout.
 	_, err := io.Copy(gw.writer, &gw.buff)
 	return err
 }

